sympy/sets/sets.py

```diff
@@ -1281,6 +1363,19 @@ def _measure(self):
             parity *= -1
         return measure
 
+    def _kind(self):
+        return self._union_setkind(self.args)
+
+    @staticmethod
+    def _union_setkind(args):
```

Comment:
If we're only defining and using this `_union_setkind` method here in the `Union` class then it might as well just be the `_kind` method.

Reply:
So it should be like this?
```python
    def _kind(self):
        kinds = tuple(arg.kind for arg in self.args if arg is not S.EmptySet)
        if not kinds:
            return SetKind()
        elif all(i == kinds[0] for i in kinds):
            return kinds[0]
        else:
            return SetKind(UndefinedKind)

```